weave

```diff
@@ -512,6 +512,26 @@ container_ip() {
     esac
 }
 
+# Given a container name or short ID in $1, ensure the specified
+# container is running and then print its full ID to stdout. If
+# it's not running or doesn't exist, print an error to stderr and
+# return with an indicative non-zero exit code.
+full_container_id() {
+    if ! STATUS=$(docker inspect --format='{{.State.Running}} {{.Id}}' $1 2>/dev/null) ; then
+        echo "Error: No such container: $1" >&2
+        return 1
+    fi
+    case "$STATUS" in
+        true*)
+            echo "${STATUS#true }"
+            ;;
+        *)
+            echo "$1 container is not running." >&2
+            return 2
+            ;;
+    esac
+}
+
```

Comment:
I think there's a nice pattern in this function that we could apply elsewhere (in `container_ip` at least) - if the function succeeds the result is printed to stdout (rather than being assigned to a global variable), otherwise it prints an error to stderr and `return`s an indicative exit status (rather than `exit`ing). This seems to give maximum flexibility to the caller:

* You can suppress the stderr messages if so desired by using `2>/dev/null`
* If such a function is invoked directly with `set -e` in operation, it will cause the script to exit
* You can call it and use its result conditionally with the `if RESULT=$(...); then` pattern
* You've got access to explicit failure reason through `$?`
* Doesn't rely on global variables